control-plane/controllers/resources/api-gateway-controller.go

```diff
@@ -14,14 +14,16 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	meshv2beta1 "github.com/hashicorp/consul-k8s/control-plane/api/mesh/v2beta1"
+	"github.com/hashicorp/consul-k8s/control-plane/gateways"
 )
 
 // APIGatewayController reconciles a APIGateway object.
 type APIGatewayController struct {
 	client.Client
-	Log        logr.Logger
-	Scheme     *runtime.Scheme
-	Controller *ConsulResourceController
+	Log           logr.Logger
+	Scheme        *runtime.Scheme
+	Controller    *ConsulResourceController
+	GatewayConfig gateways.GatewayConfig
```

Comment:
Just to be clear on this -- is this derived from the global helm configuration? The idea of `GatewayConfig` just seems kind of close to `GatewayClassConfig` to me, so the naming is slightly confusing.

Reply:
yeah, this is an existing struct we have that takes the gateway specific stuff from the helm chart, I'm down for renaming it as it is very similar to `GatewayClassConfig`